contracts/CLR.sol

```diff
@@ -149,24 +149,24 @@ contract CLR is
     /**
      *  @dev Mint CLR tokens by depositing LP tokens
      *  @dev Minted tokens are staked in CLR instance, while address receives a receipt token
-     *  @param inputAsset asset to mint with (0 - token 0, 1 - token 1)
-     *  @param amount asset mint amount
+     *  @param amount0 token 0 amount
+     *  @param amount1 token 1 amount
      */
-    function deposit(uint8 inputAsset, uint256 amount) external whenNotPaused {
-        require(amount > 0);
-        (uint256 amount0, uint256 amount1) = calculateAmountsMintedSingleToken(
-            inputAsset,
-            amount
-        );
+    function deposit(uint256 amount0, uint256 amount1) external whenNotPaused {
+        require(amount0 > 0 || amount1 > 0);
 
-        // Check if address has enough balance
-        uint256 token0Balance = token0.balanceOf(msg.sender);
-        uint256 token1Balance = token1.balanceOf(msg.sender);
-        if (amount0 > token0Balance || amount1 > token1Balance) {
```

Comment:
No need to have these checks anymore? Not sure how `safeTransferFrom` method handles the insufficient balance scenario.

Reply:
We can't rely on this logic anymore due to the risk of front-running.